client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -39,14 +44,37 @@ export class SimpleSiteRenameForm extends Component {
 		this.props.requestSiteRename( selectedSiteId, this.state.domainFieldValue, discard );
 	};
 
+	getDomainValidationMessage( domain ) {
+		const { translate } = this.props;
+
+		if ( ! domain.match( /^[a-z0-9]+$/i ) ) {
```

Comment:
Minor suggestion, I'd flip this to check for the presence of any invalid chars vs. checking that the whole string is valid.

`if ( domain.match( /[^a-z0-9]/i ) ) { //...`